Add support for long (dash dash) options for all command-line options

prologic opened the pull request:
##### Summary

ssia

##### Component Name

- area/daemon

##### Test Plan

```#!sh
(⎈ |netdata-production:infra)
prologic@Jamess-iMac
Fri Jun 05 14:25:56
~/Netdata/netdata
 (dash-dash-version) 0
$ ./netdata -v
netdata v1.22.1-209-g253b2f6a
(⎈ |netdata-production:infra)
prologic@Jamess-iMac
Fri Jun 05 14:26:04
~/Netdata/netdata
 (dash-dash-version) 0
$ ./netdata --version
netdata v1.22.1-209-g253b2f6a
(⎈ |netdata-production:infra)
prologic@Jamess-iMac
Fri Jun 05 14:26:07
~/Netdata/netdata
 (dash-dash-version) 0
$ ./netdata --help

 ^
 |.-.   .-.   .-.   .-.   .  netdata
 |   '-'   '-'   '-'   '-'   real-time performance monitoring, done right!
 +----+-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+--->

 Copyright (C) 2016-2020, Netdata, Inc. <info@netdata.cloud>
 Released under GNU General Public License v3 or later.
 All rights reserved.

 Home Page  : https://netdata.cloud
 Source Code: https://github.com/netdata/netdata
 Docs       : https://learn.netdata.cloud
 Support    : https://github.com/netdata/netdata/issues
 License    : https://github.com/netdata/netdata/blob/master/LICENSE.md

 Twitter    : https://twitter.com/linuxnetdata
 Facebook   : https://www.facebook.com/linuxnetdata/


 SYNOPSIS: netdata [options]

 Options:

  -c/--config filename              Configuration file to load.
   -                        Default: /usr/local/etc/netdata/netdata.conf

  -D/--no-fork                       Do not fork. Run in the foreground.
   -                        Default: run in the background

  -d/--daemon                       Fork. Run in the background.
   -                        Default: run in the background

  -h/--help                       Display this help message.

  -P/--pid-file filename              File to save a pid while running.
   -                        Default: do not save pid to a file

  -i/--interface IP                    The IP address to listen to.
   -                        Default: all IP addresses IPv4 and IPv6

  -p/--port port                  API/Web port to use.
   -                        Default: 19999

  -s/--prefix path                  Prefix for /proc and /sys (for containers).
   -                        Default: no prefix

  -t/--internal-clock seconds               The internal clock of netdata.
   -                        Default: 1

  -u/--user username              Run as user.
   -                        Default: netdata

  -v/--version                       Print netdata version and exit.

  -W/--set-options options               See Advanced options below.


 Advanced options:

  -W stacksize=N           Set the stacksize (in bytes).

  -W debug_flags=N         Set runtime tracing to debug.log.

  -W unittest              Run internal unittests and exit.

  -W set section option value
                           set netdata.conf option from the command line.

  -W simple-pattern pattern string
                           Check if string matches pattern and exit.

  -W "claim -token=TOKEN -rooms=ROOM1,ROOM2"
                           Claim the agent to the workspace rooms pointed to by TOKEN and ROOM*.


 Signals netdata handles:

  - HUP                    Close and reopen log files.
  - USR1                   Save internal DB to disk.
  - USR2                   Reload health configuration.
```

##### Additional Information

This has been asked for a number of times by our users from various
sources (_some of which includes our IRC channel on FreeNode :D_).

__Note:__ Please review with "Ignore Whitespace".

prologic commented:
> There is too much noise in this PR to see what has changed. Even with the hide whitespace option in github there are 623 lines of changes in one of the most delicate and dangerous files in the project. I can not verify that this does not accidentally break something.

This is likely a "style clash". As I don't actively write C I have no clue what best practices are.

I generally use [vim-autoformat](https://github.com/Chiel92/vim-autoformat) to auto-format code (_which supports many code formaters_); which in turn uses [clang-format](http://clang.llvm.org/docs/ClangFormat.html)

Is there a particular tool/config that better aligns with the existing code style I can set my editor/tools to use?

ilyam8 commented:
https://github.com/prologic/netdata/blob/485d2a5d0ce903ba34169ac515b2787bf50625c7/daemon/main.c#L60-L68

the formatting! 😄

prologic commented:
> https://github.com/prologic/netdata/blob/485d2a5d0ce903ba34169ac515b2787bf50625c7/daemon/main.c#L60-L68
> 
> the formatting! 😄
> 
> > As I don't actively write C I have no clue what best practices are.
> 
> ofc it is `c fmt ./...`!

Are you troll'n I really can't tell this time :P

amoss commented:
Unfortunately not. The code is a mismash of styles and we have not run a linting pass over it. Can you set your editor to not auto-format files instead?

prologic commented:
Okay hopefully this verison is better;

```#!sh
$ git chg
e4603722 (HEAD -> dash-dash-version) Add support for long (dash dash) options for all command-line options
 daemon/main.c | 687 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++--------------------------------------------------------------------
 daemon/main.h |   2 +
 2 files changed, 349 insertions(+), 340 deletions(-)
```

I've disabled all auto-formatting in my editor ro c/cpp.

What I did:

- Moved all the logic for handling existing options into the new `parse_args()` function borrowed from `collectors/ebpf/ebpf.c` (_Thank you @thiagoftsm_)
- Introduced a `long_name` field in the `option_def` struct.
- Added long option names to the help struct
- Added the `%c/--%s` mod to the help output
- Moved some initiationl variable to the top in the global scope so they can be used by both `parse_args()` and `main()
- Swapped all `return x;` for `exit(x);` in the moved/modified handling of existing options.
